src/session.rs

```diff
@@ -95,21 +100,30 @@ impl Session {
     /// };
     /// ```
     pub fn new(options: Options) -> AMQPResult<Session> {
-        let connection = try!(get_connection(&options));
+        let connection = get_connection(&options)?;
         let channels = Arc::new(Mutex::new(HashMap::new()));
         let (channel_zero_sender, channel_receiver) = sync_channel(CHANNEL_BUFFER_SIZE); //channel0
         let channel_zero = channel::Channel::new(0, channel_receiver, connection.clone());
-        try!(channels.lock().map_err(|_| AMQPError::SyncError)).insert(0, channel_zero_sender);
+        channels.lock().map_err(|_| AMQPError::SyncError)?.insert(0, channel_zero_sender);
         let con1 = connection.clone();
-        let channels_clone = channels.clone();
+        let channels_clone = Arc::clone(&channels);
         thread::spawn(|| Session::reading_loop(con1, channels_clone));
+
+        // If the caller has configured heartbeats, then begin the heartbeat loop.
+        if options.heartbeat > 0 {
+            let conn = connection.clone();
+            let chans = Arc::clone(&channels);
```

Comment:
The `chans` var is never used. Is it created on purpose or is it a leftover from the development?